### PR TITLE
Limetorrents fix + Unpack directory new feature

### LINKED ISSUE #1910: Limetorrent search no longer works. Invalid URL.
Branch/Commit: master/946f6be31a26c661a5e0903c5d66e4ce348ce19d
OS: Debian 7.10
What you did: Attempted to use LimeTorrent as a search provider
What happened: Search failed
What you expected: Search to succeed
Logs:

```
SEARCHQUEUE-MANUAL-277165 :: [LimeTorrents] :: GET URL: https://www.limetorrents.cc/searchrss/20/Silicon%20Valley%20S03E02/ [Status: 200]
SEARCHQUEUE-MANUAL-277165 :: [LimeTorrents] :: No data returned from provider
```

Copy/paste that URL in to browser results in empty result; blank page. Even the test case located in `tests/sickrage_tests/providers/torrent/cassettes/limetorrents.yaml` fails (https://limetorrents.cc/searchrss/20/Game%20of%20Thrones%20S05/)

The addition of `/20` appears to be the issue. Remove this and it works again.

### LINKED ISSUE #3195: unrar to temporary folder
Branch/Commit: master
OS: Ubuntu

My post processing dir (where my download client puts the downloads) is read-only, so only the Copy action makes sense. Some folders hold rar files, and since the rar files are expanded in their original location, unrar will fail as shown in the log:

```
AA No such file or directory
AA Cannot create /mnt/remoto/mirror/tv/Man.with.a.Plan.S01E14.720p.HDTV.X264-DIMENSION/man.with.a.plan.114.720p-dimension/Man.with.a.Plan.S01E14.720p.HDTV.X264-DIMENSION.mkv
AA RarNoFilesError: No files that match pattern were found [10]: 
AA raise exc(msg)
AA File "/home/sickrage/SickRage/lib/rarfile/__init__.py", line 2868, in check_returncode
AA check_returncode(p, output)
AA File "/home/sickrage/SickRage/lib/rarfile/__init__.py", line 887, in _extract
AA self._extract(fnlist, path, pwd)
AA File "/home/sickrage/SickRage/lib/rarfile/__init__.py", line 820, in extractall
AA rar_handle.extractall(path=rar_extract_path)
AA File "/home/sickrage/SickRage/sickbeard/processTV.py", line 371, in unrar
AA Traceback (most recent call last):
2017-02-22 02:02:52 ERROR POSTPROCESSOR-AUTO :: [59ce825] Failed to extract the archive man.with.a.plan.114.720p-dimension.rar: Invalid Rar Archive Usage
```

Would it be possible to have a setting to define a temporary R/W folder where unrar should take place?

Thanks for such an awesome tool!

### PR DESCRIPTION
**Pull request by sharkykh**
Proposed changes in this pull request:
- Fix Limetorrents search url to account for tv category changing from /20/ to /tv/. (Fixes #1910)
Proof: [**OLD**](https://www.limetorrents.cc/searchrss/20/Game%20of%20Thrones%20S05/) returns nothing,  [**NEW**](https://www.limetorrents.cc/searchrss/tv/Game%20of%20Thrones%20S05/) works.
RSS url still works but looks like they changed the site to move from /20/ to /tv/, so why not change it too to avoid issues again later.
- New feature: Unpack directory - choose where the unrar tool will unpack the files, in case you don't want/can't have it unrar in your tv download dir. (Fixes/Closes #3195)

The new feature was tested, but you can never be **too** careful... Test it too!

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by miigotu** [comment]
Why try to create the unpack dir? It should be an already existing dir they pick from the filebrowser.  I think you only want to check that a dir can be created in it?

**Comment by miigotu** [comment]
A for the build failing, you have to rebuild the cassette for limetorrents.
delete limetorrents.yaml, change overwrite_cassettes = False to true, run tox to recreate the yaml, change overwrite_cassettes = True back to false, commit changed limetorrents.yaml

**Comment by sharkykh** [comment]
I wasn't sure if unrar would fail if the path was incorrect, and I wanted the logs to show if it switched to the normal path.
About checking if a dir can be created, I didn't see the same check for `path`..
If it's unnecessary I'll remove that check, I'm all for that.

Trying with tox..

**Comment by sharkykh** [comment]
> Why try to create the unpack dir? It should be an already existing dir they pick from the filebrowser.
**_I think you only want to check that a dir can be created in it?_**

Well, we can do that with [helpers.makeDir, just like in change_unpack_dir](https://github.com/SickRage/SickRage/pull/3199/commits/25903dd801122497f0a9da1925722d976b7f9de3#diff-5467e5e1640f70d1d81fc06e41f39592R224), question remains if it's necessary.

File: sickbeard/config.py [hunk 1]
```diff
@@ -209,6 +209,26 @@ def change_tv_download_dir(tv_download_dir):
...
+
+    :param unpack_dir: New unpack directory
+    :return: True on success, False on failure
+    """
+    if unpack_dir == '':
+        sickbeard.UNPACK_DIR = ''
+        return True
+
+    if ek(os.path.normpath, sickbeard.UNPACK_DIR) != ek(os.path.normpath, unpack_dir):
+        if helpers.makeDir(unpack_dir):
```
Comment (miigotu) [inline sickbeard/config.py]: if isDir(unpack_dir

File: sickbeard/webserve.py [hunk 2]
```diff
@@ -4296,6 +4296,9 @@ def savePostProcessing(self, kodi_data=None, kodi_12plus_data=None,
         else:
             sickbeard.UNPACK = config.checkbox_to_value(unpack)
 
+        if not config.change_unpack_dir(unpack_dir):
+            results += ["Unable to create directory " + ek(os.path.normpath, unpack_dir) + ", dir not changed."]
```
Comment (miigotu) [inline sickbeard/webserve.py]: results += ["Unpack directory does not exist:" + ek(os.path.normpath, unpack_dir) + ", dir not changed."]

**Comment by miigotu** [comment]
Yeah in change_unpack_dir, you dont need to create the dir. The dir is a dir you selected with a filebrowser so the directory already exists. You just need to check that it is a directory and possibly check if it is writable onchange.

File: sickbeard/config.py (hunk 1 above)
Comment (sharkykh) [inline sickbeard/config.py:224]: perhaps `os.path.isdir`?
...
I mean : `ek(os.path.isdir, unpack_dir)`

Comment (miigotu) [inline sickbeard/config.py]: Basically, just for some reason I think there is a helper method to check permissions of a dir. I think we use it when changing tv_download_Dir.

Comment (miigotu) [reply sickbeard/config.py:224]: Nevermind.....

helpers.makeDir just returns True if it already exists, so this is ok

**Comment by miigotu** [comment]
Sorry lol, it was the first thing I looked at when I woke up this morning haha.

**Comment by sharkykh** [comment]
I think it was the function name that fooled ya

**Comment by sharkykh** [comment]
Whoops!
https://github.com/SickRage/SickRage/commit/8230baa81314357d94b2e3aedd02a79d3f1999dd#diff-5467e5e1640f70d1d81fc06e41f39592R214

**Comment by miigotu** [comment]
nbd, just change it in your next PR

**Pull request merged**